src/xb-silo-query.h

```diff
@@ -18,15 +19,27 @@ GPtrArray	*xb_silo_query			(XbSilo		*self,
 						 const gchar	*xpath,
 						 guint		 limit,
 						 GError		**error);
+
 GPtrArray	*xb_silo_query_full		(XbSilo		*self,
 						 XbQuery	*query,
 						 GError		**error);
+GPtrArray	*xb_silo_query_with_context	(XbSilo		*self,
+						 XbQuery	*query,
+						 XbQueryContext	*context,
+						 GError		**error);
+
 XbNode		*xb_silo_query_first		(XbSilo		*self,
 						 const gchar	*xpath,
 						 GError		**error);
+
 XbNode		*xb_silo_query_first_full	(XbSilo		*self,
 						 XbQuery	*query,
 						 GError		**error);
+XbNode		*xb_silo_query_first_with_context(XbSilo	*self,
+						 XbQuery	*query,
+						 XbQueryContext	*context,
+						 GError		**error);
+
```

Comment:
I'm wondering if we need this at all; if you're using xb_silo_query_with_context() rather than xb_silo_query_full() then it's just one _more_ line to set the limit.

Reply:
I’m afraid we do, because it’s got a different return type. I think that’s worth the additional API.

Reply:
okay, fair enough, ta.